smart_campus/app/views.py

```diff
@@ -1359,3 +1361,56 @@ def group_delete_page(request, pk):
     group_instance.delete()
 
     return HttpResponseRedirect('/groups/')
+
+
+@require_GET
+@csrf_exempt
+def get_unanswered_question(request):
+    station_id = request.GET.get('station_id')
+    email = request.GET.get('email')
+
+    station = Station.objects.filter(id=station_id).first()
+    user = User.objects.filter(email=email).first()
+
+    if not user or not station:
+        return HttpResponse('Either user or station does not exist', status=400)
+
+    unanswered_questions = Question.objects.exclude(user__pk=email).filter(
+        linked_station=station
+    )
+
+    random_index = random.randint(0, unanswered_questions.count() - 1)
+    random_unanswered_question = unanswered_questions[random_index]
+
+    choices = []
+    for i, choice in enumerate(Choice.objects.filter(question=random_unanswered_question)):
+        choices.append(choice.content)
+        if choice.is_answer:
+            index = i + 1
+
+    return JsonResponse(
+        data={
+            'content': random_unanswered_question.content,
+            'choices': choices,
+            'answer': index,
+            'question_id': random_unanswered_question.id
+        },
+        status=200
+    )
+
+
+@require_POST
+@csrf_exempt
+def add_answered_question(request):
+    question_id = request.POST.get('question_id')
+    email = request.POST.get('email')
```

Comment:
I suggest that you change the key to `user_email` for readability.

Reply:
It sounds great, I will rename them.